src/main/java/com/airlines/dao/impl/CrewMemberDaoImpl.java

```diff
@@ -23,38 +25,66 @@ public CrewMemberDaoImpl(DataSource dataSource) {
-            insertStatement.setString(1, crew.getFirstName());
-            insertStatement.setString(2, crew.getLastName());
-            insertStatement.setObject(3, crew.getPosition());
-            insertStatement.setDate(4, Date.valueOf(crew.getBirthday()));
-            insertStatement.setObject(5, crew.getCitizenship());
+            insertStatement.setString(1, crewMember.getFirstName());
+            insertStatement.setString(2, crewMember.getLastName());
+            insertStatement.setObject(3, crewMember.getPosition());
+            insertStatement.setDate(4, Date.valueOf(crewMember.getBirthday()));
+            insertStatement.setObject(5, crewMember.getCitizenship());
             insertStatement.executeUpdate();
             ResultSet generatedKeys = insertStatement.getGeneratedKeys();
             if (generatedKeys.next()) {
                 long id = generatedKeys.getLong(1);
-                crew.setId(id);
+                crewMember.setId(id);
             }
         } catch (SQLException e) {
-            throw new DaoOperationException("Error saving crew: " + crew, e);
+            throw new DaoOperationException("Error saving a member of the crew: " + crewMember, e);
         }
     }
 
     @Override
-    public CrewMember findOne(Long id) {
-        return null;
-    }
-
-    @Override
-    public CrewMember link(Airplane airplane, Long id) {
-        return null;
+    public CrewMember findById(Long id) {
+        if (id == null) {
+            throw new DaoOperationException("Cannot find the member because Id is not provided");
+        }
+        CrewMember crewMember = new CrewMember();
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_BY_ID_SQL);) {
+            selectByIdStatement.setLong(1, id);
+            ResultSet resultSet = selectByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                crewMember.setId(resultSet.getLong("id"));
+                crewMember.setFirstName(resultSet.getString("first_name"));
+                crewMember.setLastName(resultSet.getString("last_name"));
+                crewMember.setPosition(Position.valueOf(resultSet.getString("position")));
+                crewMember.setCitizenship(Citizenship.valueOf(resultSet.getString("citizenship")));
+            }
+        } catch (SQLException e) {
+            throw new DaoOperationException("Failed to find a member with Id = %d" + id, e);
+        }
+        return crewMember;
     }
 
     @Override
     public CrewMember update(CrewMember crewMember) {
-        return null;
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement statement = connection.prepareStatement(UPDATE_CREW_MEMBERS_SQL)) {
+            if (crewMember.getId() == null) {
+                throw new IllegalArgumentException("Cannot update the member because Id is not provided");
```

Comment:
change to -> "CrewMember id should not be null"

IllegalArgumentException usually focused on the argument itself, if we need add more business context (details about update operation) usually it's a task for custom exception